File: leetcode/java/2785_sort_vowels_in_a_string.java
/**
 * 2785. Sort Vowels in a String
 * Problem Link: https://leetcode.com/problems/sort-vowels-in-a-string/
 * Solution Link (Personal):
 */


class Solution {
  public String sortVowels(String s) {
    char[] chars = s.toCharArray();
    
    int idx = 0;
    char[] vowels = new char[s.length()];
    for (int i = 0; i < chars.length; i++) {
      char ch = chars[i];
      if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' 
          || ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U') {
        vowels[idx++] = ch;
      } 
    }
    
    Arrays.sort(vowels, 0, idx);
    
    idx = 0;
    for (int i = 0; i < chars.length; i++) {
      if (chars[i] == 'a' || chars[i] == 'e' || chars[i] == 'i' || chars[i] == 'o' || chars[i] == 'u'
          || chars[i] == 'A' || chars[i] == 'E' || chars[i] == 'I' || chars[i] == 'O' || chars[i] == 'U') {
        chars[i] = vowels[idx++];  
      }
    }
    
    return new String(chars);
  }
}

File: 00_leetcode/python3/2785_sort_vowels_in_a_string.py
"""
 * 2785. Sort Vowels in a String
 * Problem Link: https://leetcode.com/problems/sort-vowels-in-a-string/
"""

class Solution:
  def sortVowels(self, s: str) -> str:
    chars = list(s)
    
    vowels = []
    for i, ch in enumerate(chars):
      if ch in 'aeiouAEIOU':
        vowels.append(ch)
        chars[i] = ''
        
    vowels.sort()
    
    j = 0
    for i, ch in enumerate(chars):
      if ch == '':
        chars[i] = vowels[j]
        j += 1
        
    return ''.join(chars)